app/routes/task_routes.py

```diff
@@ -0,0 +1,135 @@
+
+# ==================================
+# Get all tasks  
+# ==================================
+@task_bp.route("", methods=["GET"])
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query == 'asc':
+        all_tasks = Task.query.order_by(Task.title.asc())
+    elif sort_query == 'desc':
+        all_tasks = Task.query.order_by(Task.title.desc())
+    else:
+        all_tasks = Task.query.all()
+    
+    results_list = []
+    for task in all_tasks:
+        results_list.append(task.to_json())
+
+    return jsonify(results_list), 200
+
+# ==================================
+# Get one task by id number
+# ==================================
+@task_bp.route("/<task_id>", methods=["GET"])
+def get_one_task(task_id):
+    return jsonify({"task": validate_id(Task, task_id).to_json()}), 200
+
+# ==================================
+# Update one task 
+# ==================================
+@task_bp.route("/<task_id>", methods=["PUT"])
+def update_one_task(task_id):
+    request_body = request.get_json()
+    task = validate_id(Task, task_id)
+    task.update(request_body)
+
+    db.session.commit()
+
+    return jsonify({"task": task.to_json()}), 200
+
+# ==================================
+# Delete one task by id
+# ==================================
+@task_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = validate_id(Task, task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    return make_response({"details":f'Task {task.task_id} "{task.title}" successfully deleted'}, 200)
+
+# ==================================
+# update one task's completeness
+# ==================================
+@task_bp.route("/<task_id>/<mark>", methods=["PATCH"])
+def mark_tasks_complete_or_incomplete(task_id, mark):
+    task = validate_id(Task, task_id)
+    task.completed_at = datetime.now() if mark == "mark_complete" else None
+    send_to_slack(task.title, "task-notifications", mark)
+    db.session.commit()
+    return jsonify({"task": task.to_json()}), 200
+
+# ==================================
+# Helper function to validate id
+# ==================================
+def validate_id(class_name,id):
+    try:
+        id = int(id)
+    except:
+        abort(make_response({"message":f"Id {id} is an invalid id"}, 400))
+
+    query_result = class_name.query.get(id)
+    if not query_result:
+        abort(make_response({"message":f"Id {id} not found"}, 404))
+
+    return query_result
+
+# ==================================
+# Helper function to send message to slack
+# ==================================
+def send_to_slack(title, channel_name, mark):
+    header_data = {'Authorization': f"Bearer {os.environ.get('SLACK_BOT_TOKEN')}"}
+    message_data = {'channel': channel_name, 'text': f"Someone just completed the task {title}"}
+    if mark == 'mark_complete':
+        requests.post('https://slack.com/api/chat.postMessage', data=message_data, headers=header_data)
```

Comment:
While I made a comment about the `validate_id` function, this `send_to_slack` function is only used by the `task_routes` so it's fine to remain here.

If you began expanding the Slack integration to `goal_routes`, then it may make sense to move this to its own module, perhaps something like `slack_utils`...